j2cl-build/src/main/java/com/vertispan/j2cl/build/provided/ClosureTask.java

```diff
@@ -153,6 +153,9 @@ public void execute(TaskOutput output) throws Exception {
             @Override
             public void finish(TaskOutput taskOutput) throws IOException {
                 Path webappDirectory = config.getWebappDirectory();
+                if(webappDirectory==null) {
+                    throw new IllegalArgumentException("No webapp directory defined");
```

Comment:
I'm curious how you encountered this - also, I think this probably should be thrown from the config.getWebappDirectory method itself, rather than modify each final task to do a null check.

Reply:
I checked out an older sample project using <defaultWebappDirectory> and got really useless NPEs. Only with running maven plugin in debugger could I found out the issue. Moving the check to elsewhere sounds good. Not sure if I will figure that out.

Reply:
You've actually already got a check in the right place, this should no longer be necessary.

Reply:
Ok, got it